api/actions/createChangelist.js

```diff
@@ -5,26 +5,25 @@
  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
 
+const Gus = require('../services/Gus');
+const { convertUrlToGusFormat } = require('./convertUrlToGusFormat');
 const GithubEvents = require('../modules/GithubEvents');
 const Issues = require('../services/Issues');
 
 module.exports = {
     eventName: GithubEvents.events.PULL_REQUEST_CLOSED,
     fn: async function(req) {
         const {
-            pull_request: { title, url, closed_at }
+            pull_request: { title, url }
         } = req.body;
         if (title.includes('@W-')) {
-            var workItemName = 'W-'.concat(title.split('@W-')[1].slice(0, -1));
-            var issue = await Issues.getByName(workItemName);
-            sails.hooks['changelists-hook'].queue.push({
-                name: 'CREATE_CHANGELIST',
-                work: issue.sfid,
-                changelist: url,
-                checkInDate: closed_at,
-                comment: url,
-                task: title
-            });
+            const workItemName = 'W-'.concat(
+                title.split('@W-')[1].slice(0, -1)
+            );
+            const issue = await Issues.getByName(workItemName);
```

Comment:
what happens when the issue with that name doesn't exists ??

Reply:
I'm now getting this directly from GUS. So, we should always get the work item Id.